bootloader/src/pyi_launch.c

```diff
@@ -467,10 +469,12 @@ int pyi_launch_execute(ARCHIVE_STATUS *status)
     if (!*PI_Py_FileSystemDefaultEncoding) {
         char *saved_locale, *loc_codeset;
         saved_locale = strdup(setlocale(LC_CTYPE, NULL));
```

Comment:
This code is actually symmetric with the one in main.c. The meaning of the `setlocale` function is a bit weird:

* `setlocale(LC_CTYPE, NULL)` means "return the current locale"
* `setlocale(LC_CTYPE, "")` means "set the current locale to the locale specified by environment variables LC_ALL, LC_CTYPE or LANG"
* `setlocale(LC_CTYPE, saved_locale)` means "set the locale to the locale in the variable saved_locale"

So at the beginning of main, it saves the starting locale (usually "C") in the variable saved_locale, and then sets the current locale to the one from the environs. Later, in run_scripts, it resets the locale to the one saved in the variable saved_locale. The log messages about LC_CTYPE should verify this.